pinot-core/src/main/java/org/apache/pinot/core/segment/creator/impl/SegmentColumnarIndexCreator.java

```diff
@@ -158,8 +158,12 @@ public void init(SegmentGeneratorConfig segmentCreationSpec, SegmentIndexCreatio
         }
 
         // Initialize inverted index creator
+        // Do not created inverted index if sorted, clean up invertedIndexColumns
         if (invertedIndexColumns.contains(columnName)) {
-          if (segmentCreationSpec.isOnHeap()) {
+          if (indexCreationInfo.isSorted()){
+            invertedIndexColumns.remove(columnName);
+            //config.getInvertedIndexCreationColumns().remove(columnName); Do we clean up config?
```

Comment:
The way the execution code is currently structured, for us to use SortedInvertedIndexBasedFilterOperator, we still need the hint for inverted-index hint - so I don't think you can remove it from config. Please trace the code and ensure that that's not broken. Also make sure there are tests covering the execution for sorted columns.

BaseSingleValueQueriesTest.java is a comprehensive test suite that should cover this, but it looks like we are not covering the case there where a column is both sorted and has invertedIndex. You can extend that test to cover this case and ensure things are working as expected. You will most likely also need a test to assert that the bitmap based index isn't generated.

Reply:
BaseSingleValueQueriesTest is for query execution. If you want to test this, you can add a separate segment generation test.

Reply:
@Jackie-Jiang, I was a requesting at test that covers the execution validation of sorted-column + inverted index as we don't seem to have any coverage for that currently.

Reply:
I tested the sorted-column + inverted index case and it did skip the inverted index generation. The HAS_INVERTED_INDEX flag is also good.